mbuild/lib/recipes/polymer.py

```diff
@@ -56,20 +79,138 @@ def __init__(self, monomers, n, sequence='A', port_labels=('up', 'down')):
                 break
 
         # Hoist the last part's top port to be the top port of the polymer.
-        self.add(last_part.labels[port_labels[0]], port_labels[0], containment=False)
+        self.add(last_part.labels[self.port_labels[0]], self.port_labels[0], containment=False)
 
         # Hoist the first part's bottom port to be the bottom port of the polymer.
-        self.add(first_part.labels[port_labels[1]], port_labels[1], containment=False)
+        self.add(first_part.labels[self.port_labels[1]], self.port_labels[1], containment=False)
+
+        # Add the end groups
+        head = self['monomer[0]'] # First monomer
+        tail = self['monomer[{}]'.format(n_monomers - 1)] # Last monomer
+        for label in self.port_labels:
+            if not head[label].used:
+                head_port = head[label]
+            if not tail[label].used:
+                tail_port = tail[label]
+
+        if not self.end_groups: # Cap each end with Hydrogens
+            hydrogen = H()
+            hydrogen['up'].update_separation(0.0547) # Defaut to H-C bond len
+            hydrogen_2 = clone(hydrogen)
+            self.end_groups.extend([hydrogen, hydrogen_2])
+            head_port.update_separation(0.0547)
+            tail_port.update_separation(0.0547)
+        else: # Use compounds in self.end_groups
+            head_port.update_separation(self.end_groups[0]['up'].separation)
+            tail_port.update_separation(self.end_groups[1]['up'].separation)
+
+        for compound in self.end_groups:
+            self.add(compound)
+        
+        force_overlap(self.end_groups[0],
+                     self.end_groups[0].labels['up'],
+                     head_port
+                     )
+        force_overlap(self.end_groups[1],
+                     self.end_groups[1].labels['up'],
+                     tail_port
+                     )
+        
+    def add_monomer(self, compound, bonding_indices, separation,
+                    port_labels=['A', 'B'], orientation=[None, None],
+                    replace=True):
+        """
+        Add an mBuild compound to self.monomers which will be used to build the polymer.
+        Call this function for each unique monomer to be used in the polymer.
+        
+        Parameters
+        ----------
+        compound : mb.Compound
+            A compound of the individual monomer
+        bonding_indices : list of int of length 2
+            The particle indicies of monomer that represent the polymer
+            bonding sites. You can specify the indices of particles that will
+            be replaced by the polymer bond, or indices of particles that act
+            as the bonding sites. See the 'replace' parameter notes.
+        separation : float, units nm
+            The bond length desired at the monomer-monomer bonding site.
+            (separation / 2) is used to set the length of each port
+        port_labels : list of str of length 2, default=['A', 'B']
```

Comment:
I agree...I was also thinking about removing the labels as a parameter all together, since it doesn't really matter what they are called (at least that I can think of). So, being able to change them wouldn't make the workflow any different or better. It might only cause issues/confusion. For example, there is a check to make sure the port labels of each monomer are the same.